homeassistant/components/sensor/dte_energy_bridge.py

```diff
@@ -16,31 +16,46 @@
 _LOGGER = logging.getLogger(__name__)
 
 CONF_IP_ADDRESS = 'ip'
+CONF_VERSION = 'version'
 
 DEFAULT_NAME = 'Current Energy Usage'
+DEFAULT_VERSION = 1
 
 ICON = 'mdi:flash'
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_IP_ADDRESS): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_VERSION, default=DEFAULT_VERSION):
+        vol.All(vol.Coerce(int), vol.Any(1, 2))
 })
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the DTE energy bridge sensor."""
     name = config.get(CONF_NAME)
     ip_address = config.get(CONF_IP_ADDRESS)
+    version = config.get(CONF_VERSION, 1)
 
-    add_devices([DteEnergyBridgeSensor(ip_address, name)], True)
+    add_devices([DteEnergyBridgeSensor(ip_address, name, version)], True)
 
 
 class DteEnergyBridgeSensor(Entity):
-    """Implementation of a DTE Energy Bridge sensor."""
+    """Implementation of the DTE Energy Bridge sensors."""
 
-    def __init__(self, ip_address, name):
+    def __init__(self, ip_address, name, version):
         """Initialize the sensor."""
-        self._url = "http://{}/instantaneousdemand".format(ip_address)
+        self._version = version
+
+        if self._version == 1:
+            url_template = "http://{}/instantaneousdemand"
+        elif self._version == 2:
+            url_template = "http://{}:8888/zigbee/se/instantaneousdemand"
+        else:
+            raise ValueError('Invalid sensor version')
```

Comment:
Remove this failback. That is not possible run into this problem

Reply:
@pvizeli Ok, it's gone.